relay_messages.go

```diff
@@ -111,7 +112,10 @@ func newLazyCallReq(f *Frame) (*lazyCallReq, error) {
 		panic(fmt.Errorf("newLazyCallReq called for wrong messageType: %v", msgType))
 	}
 
-	cr := &lazyCallReq{Frame: f}
+	cr := &lazyCallReq{
+		Frame:       f,
+		arg2Appends: make([]relay.KeyVal, 0, 2),
```

Comment:
What I was thinking was:
```
type lazyCallReq struct {
  arg2AppendInitialBuf [2]relay.KeyVal
  [..]
}

cr := &lazyCallReq {
  Frame: f,
}
cr.arg2Appends = cr.arg2AppendInitialBuf[:]
```

the current code does 2 allocations immediately, while the array approach combines the allocations

I'm curious if the array affects the performance of non-append cases. If so, we may want to skip for now (but we can do it in future if we are appending more frequently than not)